subsys/bluetooth/controller/rng_driver.c

```diff
@@ -31,11 +33,18 @@ static int rng_driver_get_entropy(struct device *dev, u8_t *buf, u16_t len)
 	u8_t *p_dst = buf;
 	u16_t bytes_left = len;
 	while (bytes_left > 0) {
-		int32_t bytes_read;
-		while ((bytes_read = ble_controller_rand_vector_get(p_dst, bytes_left)) <= 0) {
-			/* Put the thread on wait until next interrupt to get more
-			 * random values. */
-			k_sem_take(&rng_data.sem_sync, K_FOREVER);
+		int32_t bytes_read = 0;
+		while (bytes_read <= 0) {
+			if (!MULTITHREADING_LOCK_ACQUIRE()) {
```

Comment:
.. this reads "if not multithreading lock aquire" :disappointed: `== 0` for clarity? The next reader of this file will thank you :+1:

Reply:
In various files, I have just tried to maintain the same style as that of the old style. Everywhere I see, a pattern like `if (!err)` ... `if (!something)` is used. I am just continuing the same pattern. 

In the light of this comment, should we document such patterns somewhere and stick to the agreed convention? I will change it accordingly.

Reply:
Whether it should be `if (!err)` or `if (err==0)` is _not_ what I am arguing here. I am talking about how you'd read _this specific if-statement_. With the name of the function/macro, not knowing anything about what it returns, this code reads as if you're not able to acquire the lock, you should continue ;)

I'm simply asking you if you have considered this code from the point of view of someone reading this 4 weeks from now. Would you need to open `multithreading_lock.h` and read the definition of the function to understand what's going on?

Reply:
alright.